### ISSUE TITLE
v4.4.0  introduces undefined method errors

### ISSUE THREAD

**Issue by sandro-aboutsource** [thoughtbot/shoulda-matchers#1333]
After upgrading to version 4.4.0 we see errors like
```
     NoMethodError: undefined method `validate_length_of' for #<RSpec::ExampleGroups::Foo
```
in many projects. It is the same in Rails and non Rails projects.

As I see some similar issues (#1332 and #1331) it might be a general problem. Maybe it has to do with https://github.com/thoughtbot/shoulda-matchers/pull/1320?

**Comment by VSPPedro** [comment]
Thanks for sharing this!

I'll take a look.

@mcmire, I'd like to have a question cleared up, please.

I'm wondering if I should revert the changes made by the #1320 or if I should create tests to replicate the problem and then fix without removing the autoloading.

What are your thoughts on this?

**Comment by mcmire** [comment]
@VSPPedro I think for now let's just revert the commit. I took some time this morning to look into why it was that the test suite passed yet this issue occurred, and the problem is that we aren't exercising ALL of the matchers in the integration tests (we never have been, I guess). We do test a lot more of the matchers in the [tests for Shoulda](https://github.com/thoughtbot/shoulda/blob/master/test/acceptance/integrates_with_rails_test.rb), so it wouldn't be terribly difficult to copy this over and then add anything else that might be missing. However, I don't want to hold people up from taking advantage of this release, so for now let's revert.

**Comment by mcmire** [comment]
Fixed in v4.4.1!
